ggml-sycl.cpp

```diff
@@ -937,17 +961,65 @@ namespace dpct
 
     private:
         mutable std::recursive_mutex m_mutex;
+        static bool compare_dev(sycl::device &device1, sycl::device &device2)
+        {
+            dpct::device_info prop1;
+            dpct::get_device_info(prop1, device1);
+            dpct::device_info prop2;
+            dpct::get_device_info(prop2, device2);
+            return prop1.get_max_compute_units() > prop2.get_max_compute_units();
+        }
+        static int convert_backend_index(std::string & backend) {
+            if (backend == "ext_oneapi_level_zero:gpu") return 0;
+            if (backend == "opencl:gpu") return 1;
+            if (backend == "opencl:cpu") return 2;
+            if (backend == "opencl:acc") return 3;
+            printf("convert_backend_index: can't handle backend=%s\n", backend.c_str());
+            GGML_ASSERT(false);
+        }
```

Comment:
Current approach fails on NVidia and AMD targets
```suggestion
        static int convert_backend_index(std::string & backend) {
            if (backend == "ext_oneapi_level_zero:gpu") return 0;
            if (backend == "opencl:gpu") return 1;
            if (backend == "opencl:cpu") return 2;
            if (backend == "opencl:acc") return 3;
            if (backend == "ext_oneapi_cuda:gpu") return 4;
            if (backend == "ext_oneapi_hip:gpu") return 5;
            printf("convert_backend_index: can't handle backend=%s\n", backend.c_str());
            GGML_ASSERT(false);
        }
```

Reply:
Yes this is because of the gpu branches for amd/nv were not handled yet.

Reply:
I want to focus to support Intel GPU only in this PR.
I can't make sure the above code work well with other vendor GPUs, because I have no hardware environment to test other vendor GPUs.

I suggest you create new PR based on this PR to support other vendor GPUs.

Here, the index impacts the order in device list. lower is for higher priority.
I suggest change to 2,3, move CPU and ACC to 4, 5 in new PR.